### ISSUE TITLE
Add OpenBSD CI tests

### ISSUE THREAD

**Issue by aldersondrive** [bazelbuild/continuous-integration#876]
Porting Bazel to OpenBSD is underway in https://github.com/bazelbuild/bazel/issues/10250. Ideally, once the port is complete, the continuous integration tests would run for OpenBSD as well, to help ensure that the OpenBSD port remains functional.

@hlopko suggested that I file an issue on this topic in this repository. 

@philwo FYI

**Comment by philwo** [comment]
I'll look into this when I do the next round of work on our CI. :)

Things we have to find out:
[ ] How to run OpenBSD VMs on GCE?
[ ] Does Buildkite Agent work well on OpenBSD? They do provide release binaries for that platform, so I guess it's fine.

**Comment by aldersondrive** [comment]
Recognizing that CI for OpenBSD probably has low priority: It could have prevented the breakage caused by https://github.com/bazelbuild/bazel/commit/314cd3353b018420598188cf54c3f97358baa2e3, which broke the build on FreeBSD & OpenBSD.
